x/emissions/keeper/inference_synthesis/synth_palette_factory.go

```diff
@@ -27,17 +28,20 @@ func (f *SynthPaletteFactory) BuildPaletteFromRequest(req SynthRequest) SynthPal
 		AllWorkersAreNew:                 true,                            // Populated below
 		NetworkCombinedLoss:              req.NetworkCombinedLoss,
 		Epsilon:                          req.Epsilon,
+		FTolerance:                       req.FTolerance,
 		PNorm:                            req.PNorm,
 		CNorm:                            req.CNorm,
 	}
 
-	// Populates: forecastImpliedInferenceByWorker, forecasters
-	palette.UpdateForecastImpliedInferences()
-	palette.Forecasters = alloraMath.GetSortedKeys(palette.ForecastImpliedInferenceByWorker)
-
 	// Populates: infererRegrets, forecasterRegrets, allInferersAreNew, allForecastersAreNew
 	palette.BootstrapRegretData()
 	palette.AllWorkersAreNew = palette.AllInferersAreNew && palette.AllForecastersAreNew
 
-	return palette
+	// Populates: forecastImpliedInferenceByWorker,
+	err := palette.UpdateForecastImpliedInferences()
+	if err != nil {
+		return SynthPalette{}, err
+	}
+
+	return palette, nil
```

Comment:
There were a lot of bugs caused by these operations being out of order, also we should return errors